upload/common/database.py

```diff
@@ -70,3 +70,12 @@ def _run_query(query):
         engine.dispose()
         results = engine.execute(query)
     return results
+
+
+def run_query_with_params(query, params):
+    try:
+        results = engine.execute(query, params)
+    except OperationalError as e:
+        engine.dispose()
```

Comment:
Is this to reset your connections with pgbouncer?

Reply:
Yes. This protects us from situations where the client thinks it has an active connection but for some reason it is inactive. Potential causes include network issues or pgbouncer dropping inactive client connections after a long idle timeout. Sometimes its difficult to determine how long AWS actually keeps old lambda containers warmed up and waiting. This code path should not be followed often, but it is a good protective measure.

Reply:
I had questions about the "why" of this too.   Perhaps take your response above and paste it in as a code comment.